x-pack/plugins/infra/public/apps/discover_app.tsx

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { interpret } from 'xstate';
+import type { DiscoverStart } from '@kbn/discover-plugin/public';
+import type { InfraClientStartDeps, InfraClientStartExports } from '../types';
+import type { LogViewColumnConfiguration, ResolvedLogView } from '../../common/log_views';
+import { createLogViewStateMachine, DEFAULT_LOG_VIEW } from '../observability_logs/log_view_state';
+import { MESSAGE_FIELD, TIMESTAMP_FIELD } from '../../common/constants';
+
+export const renderApp = (plugins: InfraClientStartDeps, pluginStart: InfraClientStartExports) => {
+  const { discover } = plugins;
+  const { logViews } = pluginStart;
+
+  const machine = createLogViewStateMachine({
```

Comment:
Do we need to use the log view reference from the url instead of a hard-coded default here? Or am I missing some other mechanism using which it is initialized correctly?

Reply:
I didn't think of it and you are right, it was always using the default log view, even with a different one specified in the URL. I created the `initializeFromUrl` function and passed it to the state machine, now it does correctly recover the URL param if exists.